service/security/certificate/internal/src/main/java/org/eclipse/kapua/service/certificate/internal/CertificateServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+import org.eclipse.kapua.service.certificate.exception.KapuaCertificateException;
+import org.eclipse.kapua.service.certificate.internal.setting.KapuaCertificateSetting;
+import org.eclipse.kapua.service.certificate.internal.setting.KapuaCertificateSettingKeys;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@KapuaProvider
+public class CertificateServiceImpl implements CertificateService {
+
+    private X509Certificate certificate;
+    private PrivateKey privateKey;
+    public static final Logger LOGGER = LoggerFactory.getLogger(CertificateServiceImpl.class);
+    private static final KapuaLocator LOCATOR = KapuaLocator.getInstance();
+    private static final AuthorizationService AUTHORIZATION_SERVICE = LOCATOR.getService(AuthorizationService.class);
+    private static final CertificateFactory CERTIFICATE_FACTORY= LOCATOR.getFactory(CertificateFactory.class);
+    private static final PermissionFactory PERMISSION_FACTORY = LOCATOR.getFactory(PermissionFactory.class);
+    private static final CertificateDomain CERTIFICATE_DOMAIN = new CertificateDomain();
+
+    /**
+     * Constructor
+     */
+    public CertificateServiceImpl() throws KapuaException {
+        KapuaSecurityUtils.doPrivileged(() -> {
+            AUTHORIZATION_SERVICE.checkPermission(PERMISSION_FACTORY.newPermission(CERTIFICATE_DOMAIN, Actions.write, KapuaId.ANY));
+            KapuaCertificateSetting setting = KapuaCertificateSetting.getInstance();
+
+            String privateKeyPath = setting.getString(KapuaCertificateSettingKeys.CERTIFICATE_JWT_PRIVATE_KEY, "");
+            String certificatePath = setting.getString(KapuaCertificateSettingKeys.CERTIFICATE_JWT_CERTIFICATE, "");
+
+            if (privateKeyPath.isEmpty() && certificatePath.isEmpty()) {
+                // Fallback to generated
+                LOGGER.error("No private key and certificate path specified.\nPlease set authentication.session.jwt.private.key and authentication.session.jwt.certificate system properties.");
+                throw new KapuaCertificateException(KapuaCertificateErrorCodes.CERTIFICATE_ERROR);
+            } else {
+                File certificateFile = new File(certificatePath);
+                File privateKeyFile = new File(privateKeyPath);
+                certificate = CertificateUtils.readCertificate(certificateFile);
+                privateKey = CertificateUtils.readPrivateKey(privateKeyFile);
+            }
+        });
+    }
+
+    @Override
+    public Certificate create(CertificateCreator creator) throws KapuaException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Certificate find(KapuaId scopeId, KapuaId entityId) throws KapuaException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CertificateListResult query(KapuaQuery<Certificate> query) throws KapuaException {
+        Certificate kapuaCertificate = new CertificateImpl(query.getScopeId());
```

Comment:
Please add `ArgumentValidator` checks for `null` query and `null` query.getScopeId().
Please add the `AUTHORIZATION_SERVICE.checkPermission(...)` to check that the user has `certificate:view` permission